### PR TITLE
Sorting recordings 2: Integrated the UI to the recordings API.

### PR DESCRIPTION
**Pull request by KH-Amir-TN**
## Description

Enable users to sort recordings by name, length and visibility

This PR completes **2.** 
--
~~0. Extend the recordings API to sort the recordings list as indicated by the client app.~~
~~1. Update the Recordings table UI to show proper icons and handle click events.~~

At this point:
> Clicking any of the columns sort by icon should:
	~~- Show proper feedback i.e. activate the icon.~~
~~2. Connect the client app to the API and further extend the UI and mutations:~~

At this point:
> Clicking any of the columns sort by icon should:
	~~- Show proper feedback i.e. activate the icon.~~
	~~- Invalidate the recordings query cache.~~
	~~- Request a new list sorted by the selected column.~~

> Clicking attempts and outcomes:
   ~~- First click order the list by the selected column ASC.~~
   ~~- Second click order the list by the selected column DESC.~~
   ~~- Third click resets the list (No sorting).~~

~~>Note: Users can sort by multiple columns at once.~~

TODO:
   - Improve UI consistency [changing `serachParams` should change UI state, ...].
   - Revisit the `axios` config merging logic.
 
### User story [Sorting recordings]:
---
0. User login.
1. User open All recordings or a room recordings page.
2. User will have a list of the recordings.
3. User can clicks on the columns (date, recording name, room name, length, number of users, visibility, formats.) to sort the list by.
4. User will observe adequate feedbacks and the expected result.

## Testing Steps

> 1. Pull the code.
> 2. Install the dependencies `bundle install && npm|yarn install`.
> 3. Clean the previous assets build by running `rm app/assets/builds/*` (This won't remove .keep since it's hidden).
> 4. Clean the database and tmp files for a better isolation by running `rails tmp:clear && rails db:schema:cache:clear && rails db:drop && rails db:create && rails db:migrate:with_data`
> 5. Run the linter and specs `bundle exec rubocop --parallel && bundle exec rspec && npx eslint app/javascript/* --ext .jsx,.js`
> 6. Run `./bin/dev` to run the assets builders processes and the Puma server all at once.
## Screenshots (if appropriate):

image
x (1)

### CODE REVIEW

**Comment by KH-Amir-TN** [comment]
# DEPENDS ON #3566

File: app/javascript/helpers/Axios.jsx [hunk 1]
```diff
@@ -30,4 +61,11 @@ const axiosInstance = axios.create(
   },
 );
 
+// Intercepts requests, filters out and forward search params to the API.
+axiosInstance.interceptors.request.use((_config) => {
+  const params = new URLSearchParams(window.location.search);
+  mergeConfigParams(filterParams(params), _config.params);
+  return { ..._config, ...{ params } };
+}, (error) => Promise.reject(error));
+
```
Comment (farhatahmad) [inline app/javascript/helpers/Axios.jsx:70]: I think these changes need to be extracted into their own PRs with an explanation as to what and why these changes are needed. It's hard to follow a PR that makes sudden changes like this without any information

Comment (KH-Amir-TN) [reply app/javascript/helpers/Axios.jsx:71]: Details are given in #3586

File: app/javascript/hooks/queries/recordings/useRecordings.jsx [hunk 2]
```diff
@@ -1,16 +1,11 @@
...
-    headers: {
-      'Content-Type': 'application/json',
-      Accept: 'application/json',
-    },
-  }).then((resp) => {
-    setRecordings(resp.data.data);
-  }));
+export default function useRecordings(input = '') {
+  const [params] = useSearchParams();
+  params.set('search', input);
```
Comment (farhatahmad) [inline app/javascript/hooks/queries/recordings/useRecordings.jsx:7]: Is this required? Can we not just manually specificy `params: { search: input }` as we do above?

**Comment by farhatahmad** [comment]
@KH-Amir-TN Need to resolve the conflicts please

File: app/javascript/hooks/queries/recordings/useRecordings.jsx (hunk 2 above)
Comment (KH-Amir-TN) [reply app/javascript/hooks/queries/recordings/useRecordings.jsx:7]: I don't like it either, I'm changing it to something else.
